p2p/peer.go

```diff
@@ -278,9 +278,13 @@ func (p *Peer) SendHandshake(info *HandshakeInfo) error {
 
 	tstamp := eos.Tstamp{Time: info.HeadBlockTime}
 
-	signature := ecc.Signature{
-		Curve:   ecc.CurveK1,
-		Content: make([]byte, 65, 65),
+	data := []byte{
+		byte(ecc.CurveK1),
+	}
+	data = append(data, make([]byte, 65, 65)...)
+	signature, err := ecc.NewSignatureFromData(data)
```

Comment:
The `NewSignatureFromData(data)` does this internally:

```
	signature := Signature{
		Curve:   CurveID(data[0]), // 1 byte
		Content: data[1:],         // 65 bytes for K1 and R1, variable length for WA
	}
```

Which is equivalent to the previous as far as I can tell. So what is being fixed here exactly? I'm not sure I follow.

Reply:
> The `NewSignatureFromData(data)` does this internally:
> 
> ```
> 	signature := Signature{
> 		Curve:   CurveID(data[0]), // 1 byte
> 		Content: data[1:],         // 65 bytes for K1 and R1, variable length for WA
> 	}
> ```
> 
> Which is equivalent to the previous as far as I can tell. So what is being fixed here exactly? I'm not sure I follow.

@maoueh without this fix I'm getting an error
```
unable to encode message handshake ...... invalid signature: the inner signature structure must be present, was nil
```
The problem is either here or in the encoder I guess